js/aax.js

```diff
@@ -2037,6 +2039,214 @@ module.exports = class aax extends Exchange {
+        //        "currentQty": "2",
+        //        "funding": "-0.04827355",
+        //        "fundingStatus": null,
+        //        "id": "385839395735639395",
+        //        "leverage": "5",
+        //        "liquidationPrice": "2983.07",
+        //        "marketPrice": "3731.84",
+        //        "openTime": "2021-12-31T18:57:25.930Z",
+        //        "posLeverage": "5.00",
+        //        "posMargin": "14.85376824",
+        //        "quote": "USDT",
+        //        "realisedPnl": "-0.07792179",
+        //        "riskLimit": "10000000",
+        //        "riskyPrice": "3272.25",
+        //        "settleType": "VANILLA",
+        //        "stopLossPrice": "0",
+        //        "stopLossSource": 1,
+        //        "symbol": "ETHUSDTFP",
+        //        "takeProfitPrice": "0",
+        //        "takeProfitSource": 1,
+        //        "unrealisedPnl": "0.51620000",
+        //        "userID": "3829384",
+        //        "ts": 1641027194500
+        //    }
+        //
+        const contract = this.safeString (position, 'symbol');
+        market = this.safeMarket (contract, market);
+        const size = this.safeString (position, 'currentQty');
+        let side = undefined;
+        if (Precise.stringGt (size, '0')) {
+            side = 'buy';
+        } else if (Precise.stringLt (size, '0')) {
+            side = 'sell';
+        }
+        // const maintenanceRate = this.safeString (position, 'maintenance_rate');
+        // const notional = this.safeString (position, 'value');
+        const leverage = this.safeString (position, 'leverage');
+        const unrealisedPnl = this.safeString (position, 'unrealisedPnl');
+        const realisedPnl = this.safeString (position, 'realisedPnl');
+        // Initial Position Margin = ( Position Value / Leverage ) + Close Position Fee
+        const takerFee = this.fees.trading.taker;
+        // const feePaid = Precise.stringMul (takerFee, notional);
+        const initialMarginString = this.safeString (position, 'posMargin');
+        const percentage = Precise.stringMul (Precise.stringDiv (unrealisedPnl, initialMarginString), '100');
+        const timestamp = this.safeInteger (position, 'ts');
+        return {
+            'info': position,
+            'symbol': this.safeString (market, 'symbol'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'initialMargin': this.parseNumber (initialMarginString),
+            // 'initialMarginPercentage': this.parseNumber (Precise.stringDiv (initialMarginString, notional)),
+            // 'maintenanceMargin': this.parseNumber (Precise.stringMul (maintenanceRate, notional)),
+            // 'maintenanceMarginPercentage': this.parseNumber (maintenanceRate),
+            'entryPrice': this.safeNumber (position, 'avgEntryPrice'),
+            // 'notional': this.parseNumber (notional),
+            'leverage': parseInt (leverage),
```

Comment:
need to use safeNumber here

Reply:
I use `safeString` a number of lines up, because on other exchanges it's used in string math, and I thought I might need it here also. I can change it to `safeNumber` if it's not used in any math, but I won't know that until this method is done, so I'm just going to leave this for now